### PR TITLE
Debug extensions

### PR DESCRIPTION
**Pull request by frankschlegel**
Some useful extensions and APIs for easier debugging with Core Image.

### CODE REVIEW

File: Sources/DebugExtensions/DebugProxy+Export.swift [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    ///   - format: The image format. This influences bit depth (8-, 16-, or 32-bit) and pixel format (uint or float). Defaults to 8-bit uint.
+    ///             Note that not all formats are supported by all codecs. E.g., `jpeg` and `heif` only supports 8-bit formats.
+    ///   - quality: The quality of the exported image, i.e., the amount of compression. Only supported by `jpeg`, `heif`, and `heif10` codecs.
+    ///   - colorSpace: The color space of the exported image. By default, the Display P3 color space is used.
+    ///                 Note that it needs to match the chosen `format`, i.e., a single-channel format needs a grayscale color space.
+    func export(filePrefix: String? = nil, codec: ImageCodec = .tiff, format: CIFormat = .RGBAh, quality: Float = 1.0, colorSpace: CGColorSpace? = nil) {
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        let fileURL = FileManager.default.temporaryDirectory.appendingPathComponent("\(filePrefix)_image").appendingPathExtension(codec.fileExtension)
+        self.write(to: fileURL, codec: codec, format: format, quality: quality, colorSpace: colorSpace)
+        openShareSheet(for: [fileURL])
```
Comment (asemmo) [inline Sources/DebugExtensions/DebugProxy+Export.swift:91]: `openShareSheet` silently fell through on my Mac M1 (Ventura 13.2.1), i.e., no execution and message. Not sure if this is suppose to happen, the files previously exported were all there in a temporary app folder.

Comment (frankschlegel) [reply Sources/DebugExtensions/DebugProxy+Export.swift]: To clarify: This ☝️ was happening when running tests, which doesn't seem to support showing file pickers. It works normally when run inside an app.

File: Sources/DebugExtensions/DebugProxy+DebugPixel.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    /// A wrapper around a `Pixel` value that offers better human-readable
+    /// description and introspection of its values.
+    struct DebugPixel<Scalar: SIMDScalar>: CustomStringConvertible, CustomReflectable {
+        let value: Pixel<Float32>
+
+        init(_ value: Pixel<Float32>) {
+            self.value = value
+        }
+
+        public var r: Float32 { self.value.r }
```
Comment (maikz) [inline Sources/DebugExtensions/DebugProxy+DebugPixel.swift:16]: Can you do something with dynamic member lookup here?

File: Sources/DebugExtensions/DebugProxy+Export.swift [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+        let filePrefix = exportFilePrefix(filePrefix: filePrefix)
+        let shareItems = self.writeResults(to: FileManager.default.temporaryDirectory, with: filePrefix)
+        Task {
+            await openShareSheet(for: shareItems)
+        }
+    }
+
+}
+
+/// Opens a share sheet for exporting the given files from the application's main window.
```
Comment (maikz) [inline Sources/DebugExtensions/DebugProxy+Export.swift:117]: Looks like this only opens a share sheet on iOS 13 and a document picker on later versions. Isn't the method name misleading then?

File: Sources/DebugExtensions/DebugProxy+Export.swift [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+                case .exr: return "exr"
+                case .jpeg: return "jpeg"
+                case .heif, .heif10: return "heic"
+                case .png: return "png"
+                case .tiff: return "tiff"
+            }
+        }
+    }
+
+    private func write(to fileURL: URL, codec: ImageCodec, format: CIFormat = .RGBA8, quality: Float = 1.0, colorSpace: CGColorSpace?) {
```
Comment (maikz) [inline Sources/DebugExtensions/DebugProxy+Export.swift:163]: Is 1.0 quality a good default or could it be like 0.8? Does it even matter for debug though...

**Review by asemmo — APPROVED** [review]

**Review by maikz — APPROVED** [review]

**Pull request merged**